clientevent.go

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2017 Vector Creations Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+)
+
+// ClientEvent is an event which is fit for consumption by clients, in accordance with the specification.
+type ClientEvent struct {
```

Comment:
EDIT: Scratch that, it *is* useful because we situationally want to discard the `room_id` key (/sync doesn't have it set since it's in a higher up key, but other APIs do expose the `room_id` key when the context does not exist e.g AS API).